config/src/main/java/org/springframework/security/config/annotation/web/reactive/WebFluxSecurityConfiguration.java

```diff
@@ -49,12 +49,20 @@
 	private static final boolean isOAuth2Present = ClassUtils.isPresent(
 			REACTIVE_CLIENT_REGISTRATION_REPOSITORY_CLASSNAME, WebFluxSecurityConfiguration.class.getClassLoader());
 
-	@Autowired(required = false)
 	private List<SecurityWebFilterChain> securityWebFilterChains;
 
-	@Autowired
 	ApplicationContext context;
 
+	@Autowired(required = false)
+	void securityWebFilterChains(List<SecurityWebFilterChain> securityWebFilterChains) {
+		this.securityWebFilterChains = securityWebFilterChains;
+	}
+
+	@Autowired
+	void context(ApplicationContext context) {
```

Comment:
After having a look to the implementation, I am wondering if we could not avoid this `ApplicationContext` and just leverage regular bean autowiring which would be conceptually cleaner IMO.

Reply:
Good point - I reverted that change. I applied the same for  `ServerHttpSecurityConfiguration.beanFactory` as well so reverted that as well.

FYI, I did a force-push to start fresh.